packages/pathfinder/src/pathfinder.ts

```diff
@@ -0,0 +1,400 @@
+     *
+     * eth - btc
+     *
+     * but for larger amounts like 100 btc it might be more complicated
+     *
+     * eth - intermediate 1 - intermediate 2 - intermediate 3 - btc
+     *
+     * //Todo need to decide and implement the quantization method
+     */
+
+    private getBucket(
+        destinationAsset: stellarAsset,
+        destinationAmount: number,
+    ): number {
+        return 10000;
+    }
+
+    /***************************************************************************
+     * Stellar Interface
+     ***************************************************************************/
+
+    /**
+     * returns the exchange rate of 2 assets
+     */
+    private calculateExchangeRate = (destination_amount: number, source_amount: number): number => {
+        return (source_amount / destination_amount);
+    }
+
+    /**
+     * Properly constructs IPathInfo from StellarSdk.PaymentPathRecord
+     */
+    private constructPathInfo = (pathFoundResult: StellarSdk.PaymentPathRecord): IPathInfo => {
+        const { 
+            source_amount,
+            source_asset_code, 
+            source_asset_issuer, 
+            source_asset_type,
+            destination_amount,
+            destination_asset_type,
+            destination_asset_issuer,
+            destination_asset_code,
+            path,
+        } = pathFoundResult;
+        const source_asset = (source_asset_type !== 'native') 
+            ? {
+                code: source_asset_code,
+                issuerPublicKey: source_asset_issuer
+            } as stellarAsset
+            : {
+                code: 'XLM',
+                issuerPublicKey: ''
+            } as stellarAsset;
+        const destination_asset = (destination_asset_type !== 'native') 
+            ? {
+                code: destination_asset_code,
+                issuerPublicKey: destination_asset_issuer
+            } as stellarAsset
+            : {
+                code: 'XLM',
+                issuerPublicKey: ''
+            } as stellarAsset;
+        const dest_amount_str = parseFloat(destination_amount);
+        const bucket = this.getBucket(destination_asset, dest_amount_str);
+        const exchangeRate = this.calculateExchangeRate(dest_amount_str, parseFloat(source_amount));
+        const _path = path.map((asset: { asset_code: string, asset_issuer: string, asset_type: string}) => {
+            return { code: asset.asset_code, issuerPublicKey: asset.asset_issuer } as stellarAsset;
+        });
+        return {
+            to: destination_asset,
+            from: source_asset,
+            path: _path,
+            bucket: bucket,
+            exchangeRate: exchangeRate
+        } as IPathInfo;
+    }    
+     
+    /**
+     * Helper For below
+     * Creates a StellarSdk.Asset to be used in server calls
+     */
+    private constructStellarAsset = (asset: stellarAsset): StellarSdk.Asset => {
```

Comment:
we might need a better name for the `stellarAsset` interface. Something to different a stellar sdk asset from the stellar asset metadata we are carrying around and returning. Maybe `IStellarAssetMetadata`.

Reply:
i agree - it was a little confusing as well, i think that should suffice